src/osd/PG.cc

```diff
@@ -1579,7 +1576,14 @@ bool PG::choose_acting(pg_shard_t &auth_log_shard_id,
     dout(10) << "choose_acting failed, not recoverable" << dendl;
     return false;
   }
-
+  while (want.size() > pool.info.size) {
+    // async recovery should have taken out as many osds as it can.
```

Comment:
@xiexingguo does it make sense to not backport this PR to luminous since we don't have async recovery in luminous to make use of the  oversized want set?

Reply:
I thought luminous was EOL :-)

yeah, we can safely drop this patch from the luminous branch.